src/openrct2/ride/track_design.c

```diff
@@ -1312,7 +1312,7 @@ sint32 place_virtual_track(rct_track_td6 *td6, uint8 bl, uint8 rideIndex, sint32
     }
 
     if (bl == 3) {
-        return _trackDesignPlaceZ;
+         return _trackDesignPlaceZ - _trackDesignPlaceSceneryZ;
```

Comment:
You accidentally put a space before this return statement.
Also, I'd like to see the following comment above this statement:
```
// Change from vanilla: originally, _trackDesignPlaceSceneryZ was not subtracted
// from _trackDesignPlaceZ, causing bug #259.
```